stdlib/public/stubs/Stubs.cpp

```diff
@@ -487,6 +487,12 @@ static const char *_swift_stdlib_strtoX_clocale_impl(
     const char * nptr, T* outResult, T huge,
     T (*posixImpl)(const char *, char **, locale_t)
 ) {
+  if (strcasecmp(nptr, "snan") == 0 || strcasecmp(nptr, "+snan") == 0 ||
```

Comment:
I don't think it would have better performance characteristics and is less readable. I'd still go for:

```
const std::size_t len = std::strlen(nptr);
if ((len == 4 && strncasecmp(nptr, "snan", len) == 0) ||
    (len == 5 && (strncasecmp(nptr, "+snan", len) || strncasecmp(nptr, "-snan", len))) {
  *outResult = std::numeric_limits<T>::signaling_NaN();
  return nptr + len;
}
 ||
      (len == 5 && (nptr[0] == '+' || nptr[0] == '-') &&
        strncasecmp(nptr + 1, "snan", 4) == 0)) {
    // ...
  }
```